code/game/objects/items/weapons/storage/quickdraw_vr.dm

```diff
@@ -0,0 +1,79 @@
+// -----------------------------
+//       Quickdraw storage
+// -----------------------------
+//These items are pouches and cases made to be kept in belts or pockets to quickly draw objects from
+//Largely inspired by the vest pouches on Colonial Marines
+
+/obj/item/weapon/storage/quickdraw
+	name = "quickdraw"
+	desc = "This object should not appear"
+	icon = 'icons/obj/storage_vr.dmi'
+
+	//Quickmode
+	//When set to 0, this storage will operate as a regular storage, and clicking on it while equipped will open it as a storage
+	//When set to 1, a click while it is equipped will instead move the first item inside it to your hand
+	var/quickmode = 0
+
+/obj/item/weapon/storage/quickdraw/attack_hand(mob/user as mob)
+	if(src.loc == user) //If they aren't holding us, we do nothing special
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(quickmode)
+				var/first_item = contents[1]
+				if(first_item && !H.get_active_hand()) //Do we have anything to give you?
+					H.put_in_hands(first_item)
+					contents[1] = null

```

Comment:
Err. I'm not sure this will do what you want it to do.
Contents is a byond-managed list based on the loc of the object. The loc always has src in it's contents.
So when you do `put_in_hands(first_item)` it will fall out of the contents anyway.
`contents[1] = null` will either send the second item, now the first item, to nullspace, or just not do anything because byond may not let you modify contents like this.

Reply:
Can't remember where it was, but when I looked for examples of this in the code, they also seemed to set the object to null for whatever reason. Shall I change it?

Reply:
You should change it, yes, for the very reason you just said: you saw it somewhere else and used it, when it's incorrect.

The other one is probably not operating on contents (or if it actually is, it's doing so needlessly). I doubt you can directly edit the contents list this way, it probably either silently fails or runtimes.